### PR TITLE
nixos/no-x-libs: add ghostscript

### PR DESCRIPTION
**Pull request by SuperSandro2000**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: nixos/modules/config/no-x-libs.nix [hunk 1]
```diff
@@ -34,6 +34,7 @@ with lib;
       ffmpeg_5 = super.ffmpeg_5.override { ffmpegVariant = "headless"; };
       # dep of graphviz, libXpm is optional for Xpm support
       gd = super.gd.override { withXorg = false; };
+      ghostscript = super.ghostscript.override { cupsSupport = false; x11Support = false; };
```
Comment (Artturin) [inline nixos/modules/config/no-x-libs.nix:37]: Why no cups?

Comment (SuperSandro2000) [reply nixos/modules/config/no-x-libs.nix:37]: I think I mainly based it on this comment https://github.com/NixOS/nixpkgs/blob/master/pkgs/misc/ghostscript/default.nix#L26-L27

and I also didn't need it in my use case. 

Not sure where to draw the line tbh.

Comment (NickCao) [reply nixos/modules/config/no-x-libs.nix:37]: So it means without x11, cups adds a lot?

Comment (SuperSandro2000) [reply nixos/modules/config/no-x-libs.nix:37]: Yes, significantly. 

```
 ➜ command nix-build --expr 'with import ./. { }; (ghostscript.override { cupsSupport = true; x11Support = false; })' --out result-with-cups
 ➜ command nix-build --expr 'with import ./. { }; (ghostscript.override { cupsSupport = false; x11Support = false; })' --out-link result-without-cups
 ➜ nix path-info -Sh ./result-with-cups
/nix/store/xhylpmfkyskcn7ql3gvya20xvcdwag42-ghostscript-10.02.1  195.0M
 ➜ nix path-info -Sh ./result-without-cups
/nix/store/6x3bd72jrlr47ljnf6yfxn9im9zhq9ka-ghostscript-10.02.1   99.2M
```

**Review by NickCao — APPROVED** [review]
Result of `nixpkgs-review pr 272552` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>

**Pull request merged**